doc/releases/changelog-dev.md

```diff
@@ -106,21 +107,52 @@
     significant performance improvement when executing the QNode on remote
     quantum hardware.
 
+  - When decomposing the circuit, the default decomposition strategy will prioritize
```

Comment:
> This is beneficial in terms of a reduction in quantum resources, at the expense of moving the device decomposition down to every evaluation of the device (so additional classical overhead).

Seems to be yet another benefit for caching parametric circuits to reuse device translations.

Reply:
Yep, precisely 💯 I would even argue, this is only fully solved by parametric _compilation_.

Reply:
Followup question:  suppose I do something like 

```python
@qml.compile()
@qml.qnode(dev, expansion_strategy="device")
def some_qnode():
    # stuff
```
(or alternatively the gradient strategy). When does decomposition happen currently vs. in this new PR w.r.t. the compilation transform? As you suggest @josh146 we would want compilation to happen before either the `device` or `gradient` strategy so that compilation it is acting on a smaller circuit rather than the full-depth expanded one, and consequently leading to a smaller circuit that gets expanded / gradient transformed. (That said, it's possible that a decomposition leads to optimizations in the compilation pipeline that might not otherwise be recognized...)

Reply:
Thanks for the detailed explanation @josh146, that makes things very clear! The gradient transform continues to impress me!

Reply:
@glassnotes, correct me if wrong, but `compile()` is a qfunc transform, not a QNode transform? So the following order is needed:

```python
@qml.qnode(dev, expansion_strategy="device")
@qml.compile()
def some_qnode():
    # stuff
```

Reply:
Yes you're 100% correct, my bad :woman_facepalming: